.github/workflows/e2e.yml

```
@@ -8,6 +8,9 @@ on:
     branches:
       - main
 
+permissions:
+  contents: read
```

```suggestion
  contents: read
  actions: write
```

For uploading artifacts